app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -741,4 +747,26 @@ public void onRowClicked(int inputPosition)
             startActivityForResult(i, Utils.MAIN_REQUEST);
         }
     }
+
+    @Override
+    protected void onPostResume() {
+        super.onPostResume();
+
+        settings = new Settings(this);
+        if(settings.getAppLockStatus()){  // App Lock On
+            new UnlockDialogBuilder(this)
+                    .onUnlocked(() -> {
+                        appIsLocked = false;
+                        updateLoyaltyCardList();
+                        Toast.makeText(this, "Unlocked", Toast.LENGTH_SHORT).show();
+                    })
+                    .onCanceled(() -> { })
```

Comment:
Can we disable cancelling? Pressing backspace gets you in buggy states. We should always force unlock if the user set locking.